### PR TITLE
Fix rational-compile for issue #153

### PR DESCRIPTION
**Pull request by abarocio80**
# Fix `rational-compile` for issue #153 
Still have a strange bug with `native-compile`.

If you start your emacs session configured with `rational-compile` to
compile modules, init or config on startup, it will stall on a second
startup, complaining that `rational-package-install-package` is not
available.

The issue seems to be solved once you delete all `.elc` files from
`user-emacs-directory` and `rational-config-path`, and only then
`rational-compile-init`, `rational-compile-config` or
`rational-compile-modules`.

I don't know if that problem is specific to my system/configuration,
but I have tried daviwil's configuration with only `rational-compile`
added to it, and it crashes.

Maybe if `rational-package-install-package` is turned into a module,
and that module is required on each module that uses it, it could stop
complaining that it is not available.

Please read, comment, make suggestions and corrections.

### CODE REVIEW

**Comment by abarocio80** [comment]
Just seen issue #187. Will wait if it needs to first go through renaming process.

**Comment by ajxn** [comment]
Is it possible to detect the absence of **rational-package-install-package** and then remove the .elc files as a hack, because it seams like this problem is also in my setup.

I am using latest master.

**Comment by abarocio80** [comment]
My guess is that, as those files got compiled withouth a `rational-package-install-package` function (in the compilation context), don't know what to do when the compiled file is loaded and executed, no matter if the function is already in the execution context.

So, I would propose that each module that make use of `rational-package-install-package` require the package `rational-package` (from #191, now renamed to `crafted-package` and the function `crafted-package-install-package` according to #194).

If some one knows better than this, pelase let me know.

**Comment by jeffbowman** [comment]
@abarocio80 Do you want to take a look into the conflicts with this PR? Maybe once those are resolved we can merge it?

**Comment by abarocio80** [comment]
@jeffbowman, I think I did it.

File: modules/crafted-compile.el [hunk 1]
```diff
@@ -239,7 +245,23 @@ The files to be compiled is defined in
                        (string-equal major-mode "emacs-lisp-mode"))
               (crafted-compile-buffer))))
 
-
+(defun crafted-compile-locate-eln-file (library)
+  ""
```
Comment (jeffbowman) [inline modules/crafted-compile.el:249]: Maybe remove the empty string, or (better) provide some doc here

**Review by jeffbowman — CHANGES_REQUESTED** [review]
Thanks, minor change requested.

**Comment by abarocio80** [comment]
OK. I think it is done.

**Comment by jeffbowman** [comment]
Thanks!

**Pull request merged**